packages/x-components/src/components/panels/__tests__/base-tabs-panel.spec.ts

```diff
@@ -0,0 +1,218 @@
+  });
+
+  it('renders only the tabs list when no `initialTab` is passed', () => {
+    const { getTabPanel, getTabsButtons } = renderBaseTabsPanel({
+      tabs: {
+        summer: '<div>Top Summer sales</div>',
+        fall: '<div>Top Fall sales</div>',
+        outlet: '<div>Top Outlet sales</div>'
+      }
+    });
+
+    expect(getTabsButtons()).toHaveLength(3);
+    expect(getTabPanel().exists()).toBe(false);
+  });
+
+  it('renders only the tabs list when passed `initialTab` is invalid', () => {
+    const { getTabPanel, getTabsButtons } = renderBaseTabsPanel({
+      tabs: {
+        summer: '<div>Top Summer sales</div>',
+        fall: '<div>Top Fall sales</div>',
+        outlet: '<div>Top Outlet sales</div>'
+      },
+      initialTab: 'whatever'
+    });
+
+    expect(getTabsButtons()).toHaveLength(3);
+    expect(getTabPanel().exists()).toBe(false);
+  });
+
+  it('renders the panel of the tab indicated by `initialTab`', () => {
+    const { getTabPanel } = renderBaseTabsPanel({
+      tabs: {
+        summer: '<div>Top Summer sales</div>',
+        fall: '<div>Top Fall sales</div>'
+      },
+      initialTab: 'fall'
+    });
+
+    expect(getTabPanel().text()).toBe('Top Fall sales');
+  });
+
+  it('renders the default `tab` slot', () => {
+    const { getTabsButtons } = renderBaseTabsPanel({
+      tabs: {
+        summer: '<div>Top Summer sales</div>'
+      }
+    });
+
+    expect(getTabsButtons()).toHaveLength(1);
+    expect(getTabsButtons().at(0).text()).toBe('summer');
+  });
+
+  it('renders a custom `tab` slot properly', async () => {
+    const { clickNthTab, getTabsButtons } = renderBaseTabsPanel({
+      slots: {
+        tab: `<template v-slot="{ tab, isSelected, select }">
+            <button data-test="base-tabs-panel-button" @click="select">
+              custom {{ tab }} tab <span v-if="isSelected">✅</span>
+            </button>
+          </template>`
+      },
+      tabs: {
+        summer: '<div>Summer top sales</div>'
+      }
+    });
+
+    expect(getTabsButtons()).toHaveLength(1);
+    expect(getTabsButtons().at(0).text()).toBe('custom summer tab');
+
+    // Select first tab
+    await clickNthTab(0);
+    expect(getTabsButtons().at(0).text()).toBe('custom summer tab ✅');
+  });
+
+  it('renders a custom `tab-content` slot properly', async () => {
+    const { clickNthTab, getTabsButtons } = renderBaseTabsPanel({
+      slots: {
+        'tab-content': `<template v-slot="{ tab, isSelected }">
+            custom {{ tab }} tab content <span v-if="isSelected">✅</span>
+          </template>`
+      },
+      tabs: {
+        summer: '<div>Top Summer sales</div>'
+      }
+    });
+
+    expect(getTabsButtons()).toHaveLength(1);
+    expect(getTabsButtons().at(0).text()).toBe('custom summer tab content');
+
+    // Select first tab
+    await clickNthTab(0);
+    expect(getTabsButtons().at(0).text()).toBe('custom summer tab content ✅');
+  });
+
+  it('changes the selected tab on click', async () => {
```

Comment:
Can you add to this test some assertions to check if the `aria-selected` attribute is being updated?

Reply:
Done!